lib/ddtrace/contrib/sinatra/tracer.rb

```diff
@@ -0,0 +1,180 @@
+
+require 'sinatra/base'
+
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/errors'
+require 'ddtrace/ext/http'
+
+sinatra_vs = Gem::Version.new(Sinatra::VERSION)
+sinatra_min_vs = Gem::Version.new('1.4.0')
+if sinatra_vs < sinatra_min_vs
+  raise "sinatra version #{sinatra_vs} is not supported yet " \
```

Comment:
Yes it's meant to be a crash if users are not handling the exception 😃 
But it's just a nitpick so let's fix tests and rebase the PR.